gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -311,22 +320,41 @@ var gpii = fluid.registerNamespace("gpii");
         return value;
     };
 
+    /** Remove all the settings blocks from a solutions registry entry
+      */
+    gpii.lifecycleManager.removeSettingsBlocks = function (solutionEntry) {
+        fluid.each(["settingsHandlers", "launchHandlers"], function (handlersBlock) {
+            solutionEntry[handlersBlock] = fluid.transform(solutionEntry[handlersBlock], function (handler) {
+                return fluid.filterKeys(handler, "settings", true);
+            });
+        });
+        return solutionEntry;
+    };
+
     /** Applies snapshotted settings from a single settingsHandler block attached to a single solution into the "originalSettings"
      * model snapshot area in the LifecycleManager's session. Tightly bound to executeSettingsAction, executes one-to-one with it
      * with almost identical argument list.
      */
-
-    gpii.lifecycleManager.recordSnapshotInSession = function (that, snapshot, solutionId, solutionRecord, session, settingsHandlerBlockName, rootAction) {
-        var toSnapshot = fluid.copy(solutionRecord);
-        toSnapshot.settingsHandlers = {};
-        toSnapshot.settingsHandlers[settingsHandlerBlockName] = snapshot;
-        if (rootAction === "start") {
-            session.applier.change(["originalSettings", solutionId], toSnapshot);
-        } else if (rootAction === "update") {
-            // if we're doing an update, keep the settings that are already stored from the
+    gpii.lifecycleManager.recordSnapshotInSession = function (that, snapshot, solutionId, solutionRecord, session, handlerType, settingsHandlerBlockName, rootAction) {
+        if (rootAction === "start" || rootAction === "update") {
+            var toSnapshot = gpii.lifecycleManager.removeSettingsBlocks(fluid.copy(solutionRecord));
+
+            // Disabled the removing of all settingshandlers blocks to ensure that we have them available when restoring the system.
```

Comment:
Rather than having commenting out code lying around, we should reword the comment to simply reflect the current state - e.g. "Do not remove settingsHandler blocks, since ..."